app/components/autocomplete/channel_mention/channel_mention.js

```diff
@@ -70,7 +73,49 @@ export default class ChannelMention extends PureComponent {
         if (matchTerm !== this.props.matchTerm) {
             // if the term changed and we haven't made the request do that first
             const {currentTeamId} = this.props;
-            this.props.actions.searchChannels(currentTeamId, matchTerm);
+            if (!isMinimumServerVersion(this.props.serverVersion, 5, 4)) {
+                this.props.actions.searchChannels(currentTeamId, matchTerm);
+                return;
+            }
+            this.props.actions.autocompleteChannelsForSearch(currentTeamId, matchTerm).then((response) => {
+                if (isSearch) {
+                    const completePublicChannels = response.data.reduce((acc, val) => {
```

Comment:
should this be taken from the props?

Reply:
Yes. I did in this way because the autocompleteChannelsForSearch give exactly the results needed (and then store them in the channels). If I take it from the props I need to filter stuff in the frontend part too, and is duplication of code, but here, the code is already duplicated so... probably make sense to reuse that code.

Reply:
Using the same block of code to render in both cases.